src/components/icon/icon.registry.ts

```diff
@@ -25,6 +37,21 @@ export class IconsRegistry {
     this.references = new DefaultMap(() => new Map());
 
     this.collections.set('internal', internalIcons);
+    this.iconBroadcastChannel = new BroadcastChannel('ignite-ui-icon-channel');
+    this.iconBroadcastChannel.onmessage = (event) => {
+      const message = event.data as BroadcastIconsChangeMessage;
+      if (message.actionType === ActionType.SyncState) {
+        // send state
+        const userSetCollection: Map<string, Map<string, SvgIcon>> = this.getUserSetCollection();
+        const refs: Map<string, Map<string, IconMeta>> = this.getUserSetRefs();
+        const message: BroadcastIconsChangeMessage = {
+          actionType: ActionType.SyncState,
+          collections: userSetCollection,
+          references: refs
+        };
+        this.iconBroadcastChannel.postMessage(message);
+      }
+    };
```

Comment:
Maybe we can move the whole logic/state around the broadcast channel into a different module/class?